nessie/src/main/java/org/apache/iceberg/nessie/NessieTableOperations.java

```diff
@@ -139,15 +170,27 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       }
       Snapshot snapshot = metadata.currentSnapshot();
       long snapshotId = snapshot != null ? snapshot.snapshotId() : -1L;
+
+      JsonNode newMetadata;
+      try {
+        TokenBuffer tokenBuffer = new TokenBuffer(JsonUtil.factory().getCodec(), false);
+        TableMetadataParser.toJson(metadata, tokenBuffer);
```

Comment:
So we could avoid this usage if we use the "toJson(metadata) method and then pass that into the token buffer. I know it's a bit more wasteful but we are only doing this once per operation and it let's us keep the toJson(metadata, generator) method package private. Again I may be being paranoid here but ideally we don't open up any more methods than we absolutely have to.